src/main/thrift/parquet.thrift

```diff
@@ -595,6 +607,62 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+/** Offset of the page in the file **/
+  1: required i64 offset
+
+/**
+ * Size of the page, including header. Sum of compressed_page_size and header
+ * length
+ */
+  2: required i32 compressed_page_size
+
+/**
+ * Index within the RowGroup of the first row of the page; this means pages
+ * change on record boundaries (r = 0).
+ */
+  3: required i64 first_row_index
+}
+
+struct OffsetIndex {
+/**
+ * PageLocations, ordered by increasing PageLocation.offset. It is required
+ * that page_locations[i].first_row_index > page_locations[i+1].first_row_index.
```

Comment:
did you mean "<" ?
first_row_index should be increasing as well

Reply:
Yes